src/language-markdown/print-preprocess.js

```diff
@@ -31,7 +31,17 @@ function transformInlineCode(ast) {
       return node;
     }
 
-    return { ...node, value: node.value.replace(/\s+/g, " ") };
+    const valueWithoutNewlines = node.value.replace(/\n/g, " ");
```

Comment:
Let's preserve new lines when `proseWrap` is set to `preserve'. This PR will supersede #11373 then.

Reply:
#11373 has been merged. Please rebase.